cpp/locality/Voronoi.cc

```diff
@@ -137,9 +138,13 @@ void Voronoi::compute(const freud::locality::NeighborQuery* nq)
                 const vec3<double> normal(normals[3 * neighbor_counter], normals[3 * neighbor_counter + 1],
                                           normals[3 * neighbor_counter + 2]);
 
-                // Ignore bonds in 2D systems that point up or down
-                if (box.is2D() && std::abs(normal.z) > 0)
+                // Ignore bonds in 2D systems that point up or down. This check
+                // should only be dealing with bonds whose normal vectors' z
+                // components are -1, 0, or +1 (within some tolerance).
+                if (box.is2D() && std::abs(normal.z) > 1e-5)
```

Comment:
@vyasr and I discussed and preferred to use 0.5 as the tolerance, which ensures that we "split the difference" between 0 and 1. We hope it is a more robust choice.
```suggestion
                if (box.is2D() && std::abs(normal.z) > 0.5)
```